multisig-wallet/src/lib.rs

```diff
@@ -0,0 +1,508 @@
+            .iter()
+            .position(|x| *x == *old_owner)
+            .expect("Can't find old owner");
+        self.owners[old_owner_index] = *new_owner;
+
+        *self.is_owner.entry(*old_owner).or_default() = false;
+        *self.is_owner.entry(*new_owner).or_default() = true;
+
+        msg::reply(
+            MWEvent::OwnerReplace {
+                old_owner: *old_owner,
+                new_owner: *new_owner,
+            },
+            0,
+        )
+        .unwrap();
+    }
+
+    /// Allows to change the number of required confirmations. Transaction has to be sent by wallet.
+    /// required Number of required confirmations.
+    fn change_requirement(&mut self, required: u64) {
+        self.validate_only_wallet();
+        validate_requirement(self.owners.len(), required);
+
+        self.required = required;
+
+        msg::reply(MWEvent::RequirementChange(required.into()), 0).unwrap();
+    }
+
+    ///  Allows an owner to submit and confirm a transaction.
+    ///  destination Transaction target address.
+    ///  value Transaction ether value.
+    ///  data Transaction data payload.
+    ///  Returns transaction ID.
+    fn submit_transaction(&mut self, destination: &ActorId, data: Vec<u8>, value: u128) -> U256 {
+        let transaction_id = self.add_transaction(destination, data, value);
+        self.confirm_transaction(&transaction_id);
+
+        msg::reply(MWEvent::Submission { transaction_id }, 0).unwrap();
+
+        transaction_id
+    }
+
+    /// Allows an owner to confirm a transaction.
+    /// transactionId Transaction ID.
+    fn confirm_transaction(&mut self, transaction_id: &U256) {
+        self.validate_owner_exists(&msg::source());
+        self.validate_transaction_exists(transaction_id);
+        self.validate_not_confirmed(transaction_id, &msg::source());
+
+        let confirmation = self
+            .confirmations
+            .entry(*transaction_id)
+            .or_insert(BTreeMap::new())
+            .entry(msg::source())
+            .or_insert(true);
+
+        *confirmation = true;
+
+        self.execute_transaction(transaction_id, None::<fn()>);
+    }
+
+    fn external_confirm_transaction(&mut self, transaction_id: &U256) {
+        self.confirm_transaction(transaction_id);
+
+        msg::reply(
+            MWEvent::Confirmation {
+                sender: msg::source(),
+                transaction_id: *transaction_id,
+            },
+            0,
+        )
+        .unwrap();
+    }
+
+    /// Allows an owner to revoke a confirmation for a transaction.
+    /// transactionId Transaction ID.
+    fn revoke_confirmation(&mut self, transaction_id: &U256) {
+        self.validate_owner_exists(&msg::source());
+        self.validate_confirmed(transaction_id, &msg::source());
+        self.validate_not_executed(transaction_id);
+
+        let confirmation = self
+            .confirmations
+            .entry(*transaction_id)
+            .or_insert(BTreeMap::new())
```

Comment:
A little bit shorter:
```suggestion
            .or_default()
```